trollsift/tests/unittests/test_parser.py

```diff
@@ -320,3 +320,72 @@ def test_greediness(self):
         template = '{band_type:s}_{polarization_extracted}_{unit}_{s1_fname}'
         res_dict = parse(template, fname)
         self.assertEqual(exp, res_dict)
+
+
+class TestParserFixedPoint:
+    """Test parsing of fixed point numbers."""
+
+    @pytest.mark.parametrize(
+        'test_case',
+        [
+            # Naive
+            {'fmt': '{foo:f}', 'string': '12.34', 'expected': 12.34},
+            # Including width and precision
+            {'fmt': '{foo:5.2f}', 'string': '12.34', 'expected': 12.34},
+            {'fmt': '{foo:5.2f}', 'string': '-1.23', 'expected': -1.23},
+            {'fmt': '{foo:5.2f}', 'string': '12.34', 'expected': 12.34},
+            {'fmt': '{foo:5.2f}', 'string': '123.45', 'expected': 123.45},
+            # Whitespace padded
+            {'fmt': '{foo:5.2f}', 'string': ' 1.23', 'expected': 1.23},
+            {'fmt': '{foo:5.2f}', 'string': ' 12.34', 'expected': 12.34},
+            # Zero padded
+            {'fmt': '{foo:05.2f}', 'string': '01.23', 'expected': 1.23},
+            {'fmt': '{foo:05.2f}', 'string': '012.34', 'expected': 12.34},
+            # Only precision, no width
+            {'fmt': '{foo:.2f}', 'string': '12.34', 'expected': 12.34},
+            # Only width, no precision
+            {'fmt': '{foo:16f}', 'string': '            1.12', 'expected': 1.12},
+            # No digits before decimal point
+            {'fmt': '{foo:3.2f}', 'string': '.12', 'expected': 0.12},
+            {'fmt': '{foo:4.2f}', 'string': '-.12', 'expected': -0.12},
+            {'fmt': '{foo:4.2f}', 'string': ' .12', 'expected': 0.12},
+            {'fmt': '{foo:4.2f}', 'string': '  .12', 'expected': 0.12},
+            {'fmt': '{foo:16f}', 'string': '             .12', 'expected': 0.12},
+            # Exponential format
+            {'fmt': '{foo:7.2e}', 'string': '-1.23e4', 'expected': -1.23e4},
+        ]
+    )
+    def test_match(self, test_case):
```

Comment:
I think the pytest way of doing this would be to make `fmt`, `string`, `expected` as three separate arguments. So it would look like:

```
@pytest.mark.parameterize(
    "fmt string expected",
    [
        ('{foo:f}', '12.34', 12.34),
        ...
    ])
    def test_match(self, fmt, string, expected):
````

Reply:
Could also define the args in parameterize as a tuple `("fmt", "string", "expected")`.

Reply:
Good suggestion, thanks!